admin/src/js/controllers/edit-user.js

```diff
@@ -254,24 +277,57 @@ angular
       return hasPlace && hasContact;
     };
 
+    const validateFacilityHierarchy = () => {
+      const placeIds = $scope.editUserModel.place;
+      const docsFetched = 10;
+
+      if (!placeIds || placeIds.length === 1) {
+        return $q.resolve(true);
+      }
+
+      return DB().allDocs({ keys: placeIds, include_docs: true, limit: docsFetched })
+        .then(result => {
+          const places = result.rows.map(row => row.doc);
+          const contactTypes = new Set(places.map(place => place.contact_type));
```

Comment:
I believe that checking the `contact_type` is not enough here. 
A contact can have a `type` and a `contact_type` as well. 

Have a look at the contact type utils library: https://github.com/medic/cht-core/blob/master/shared-libs/contact-types-utils/src/index.js#L13

I think this function, that checks whether a list of contacts are of the same type can belong there, instead of here.

Reply:
I have moved the work of checking the `contact_types` to the contact type utils library. But I don't understand what you mean by 

> checking the contact_type is not enough here.

the check confirms that all the places have the same contact_type (community_health_unit), why is that not enough for ensuring they are at the same hierarchy level ?

Reply:
If you look at the code from the library, you'll see that a contact can have a `type` with a hardcoded contact type and a `contact_type` with a configurable contact type. 

Additionally, one type of contact can have multiple types of parents. so you can have a certain place at different levels of the hierarchy, but I don't think we have to worry about this ... yet.

Reply:
Sorry, I don't understand the issue (difference) with `type` and `contact_type`. I don't see where `HARDCODED_TYPES` comes in the equation. 

Since all places have the same type `contact` and the hierarchy level is defined by `contact_type`, why is the check to ensure they all have the same `contact_type`  insufficient ?

Reply:
Not all places have type `contact`.... 
Please take the time to read the code of https://github.com/medic/cht-core/blob/master/shared-libs/contact-types-utils/src/index.js#L13 to understand or read the docs: https://docs.communityhealthtoolkit.org/apps/reference/app-settings/hierarchy/#changing-the-configuration
Or look at our sample data. 

It perfectly valid to have a place that has: { type: 'clinic' } and no `contact_type`.

Reply:
Thanks for the patience with my daftness 😅 . I read the docs and added some logs, now I get what you mean. Now it's ready for review.